common/thrift_router.h

```diff
@@ -207,6 +207,31 @@ class ThriftRouter {
     return itor->second.shard_to_hosts.size();
   }
 
+  // Note: This function is AZ unaware. It returns the total number hosts
+  // within the segment and shard.
+  uint32_t getHostNumberFor(const std::string& segment, const ShardID shard) {
+    std::shared_ptr<const ClusterLayout> layout;
+    {
+      folly::RWSpinLock::ReadHolder read_guard(layout_rwlock_);
+      layout = cluster_layout_;
+    }
```

Comment:
now we have three identical code sections in this file.
`{
      folly::RWSpinLock::ReadHolder read_guard(layout_rwlock_);
       layout = cluster_layout_;
 }`
Can you refactor the code a little bit by adding a private member function getClusterLayout()?